attributecert/attributecert.go

```diff
@@ -0,0 +1,712 @@
+}
+
+type ComponentAddress struct {
+	AddressType  asn1.ObjectIdentifier
+	AddressValue string
+}
+
+type ComponentClass struct {
+	ComponentClassRegistry asn1.ObjectIdentifier
+	ComponentClassValue    string
+}
+
+type ComponentIdentifierV2 struct {
+	ComponentClass           ComponentClass
+	ComponentManufacturer    string
+	ComponentModel           string
+	ComponentSerial          string                `asn1:"optional,tag=0"`
+	ComponentRevision        string                `asn1:"optional,tag=1"`
+	ComponentManufacturerId  int                   `asn1:"optional,tag=2"`
+	FieldReplaceable         bool                  `asn1:"optional,tag=3"`
+	ComponentAddresses       []ComponentAddress    `asn1:"optional,tag=4"`
+	ComponentPlatformCert    CertificateIdentifier `asn1:"optional,tag=5"`
+	ComponentPlatformCertUri string                `asn1:"optional,tag=6"`
+	Status                   asn1.Enumerated       `asn1:"optional,tag=7"`
+}
+
+type UriReference struct {
+	UniformResourceIdentifier string
+	HashAlgorithm             pkix.AlgorithmIdentifier `asn1:"optional"`
+	HashValue                 string                   `asn1:"optional"`
+}
+
+type PlatformConfigurationV2 struct {
+	ComponentIdentifiers    []ComponentIdentifierV2 `asn1:"optional,tag=0"`
+	ComponentIdentifiersUri UriReference            `asn1:"optional,tag=1"`
+	PlatformProperties      []Property              `asn1:"optional,tag=2"`
+	PlatformPropertiesUri   UriReference            `asn1:"optional,tag=3"`
+}
+
+type ComponentIdentifierV1 struct {
+	ComponentManufacturer   string
+	ComponentModel          string
+	ComponentSerial         string             `asn1:"optional,tag=0"`
+	ComponentRevision       string             `asn1:"optional,tag=1"`
+	ComponentManufacturerId int                `asn1:"optional,tag=2"`
+	FieldReplaceable        bool               `asn1:"optional,tag=3"`
+	ComponentAddresses      []ComponentAddress `asn1:"optional,tag=4"`
+}
+
+type PlatformConfigurationV1 struct {
+	ComponentIdentifiers  []ComponentIdentifierV1 `asn1:"optional,tag=0"`
+	PlatformProperties    []Property              `asn1:"optional,tag=1"`
+	PlatformPropertiesUri []UriReference          `asn1:"optional,tag=2"`
+}
+
+func unmarshalSAN(v asn1.RawValue) ([]pkix.AttributeTypeAndValue, error) {
+	var attributes []pkix.AttributeTypeAndValue
+	if v.Tag == asn1.TagSet {
+		var e pkix.AttributeTypeAndValue
+		_, err := asn1.Unmarshal(v.Bytes, &e)
+		if err != nil {
+			return nil, err
+		}
+		attributes = append(attributes, e)
+		return attributes, nil
+	} else if v.Tag == asn1.TagOctetString {
+		var platformData PlatformDataSequence
+		rest, err := asn1.Unmarshal(v.Bytes, &platformData)
+		if err != nil {
+			return nil, err
+		} else if len(rest) != 0 {
+			return nil, errors.New("attributecert: trailing data after X.509 subject")
+		}
+		for _, e := range platformData {
+			for _, e2 := range e {
+				attributes = append(attributes, e2)
+			}
+		}
```

Comment:
For consistency with the other branch, do a `return attributes, nil` here, so the return at the close of the function only needs to handle the case where neither branch is hit.